frontend/src/modules/app/layouts/SidebarSlideRoute.tsx

```diff
@@ -112,6 +160,38 @@ const FullWidthToolBar = (): JSX.Element => {
           </Drawer>
         </React.Fragment>
       </Grid>
+      <ModalPopup
```

Comment:
Are the contents of this `ModalPopup` component copy-pasted from somewhere else? If so, we should decompose it into a separate component and/or a separate container (i.e. a component bound to selectors using react-redux's `connect`). If we have the same few modal popups copy-pasted across the frontend codebase, maybe we should make a "frontend/src/modules/Modals" folder which contains reusable modals, e.g. for submitting/discarding unsaved changes to alarm limits, and for submitting/discarding unsaved changes to parameters.

Reply:
Yes `ModalPopup` is copy-pasted multiple times, I will try to add a re-usable component in this PR.

Reply:
we will do this in a future PR